python/tests/proton_tests/main.py

```diff
@@ -330,382 +356,401 @@ def filter(self, record):
+        runner = Runner()
+        runner.run("test", lambda: self.test(config))
+        return runner
 
-  def _run(self):
-    runner = Runner()
-    runner.run("test", lambda: self.test(config))
-    return runner
+    def __repr__(self):
+        return "FunctionTest(%r)" % self.test
 
-  def __repr__(self):
-    return "FunctionTest(%r)" % self.test
 
 class MethodTest:
 
-  def __init__(self, cls, method):
-    self.cls = cls
-    self.method = method
+    def __init__(self, cls, method):
+        self.cls = cls
+        self.method = method
 
-  def name(self):
-    return "%s.%s.%s" % (self.cls.__module__, self.cls.__name__, self.method)
+    def name(self):
+        return "%s.%s.%s" % (self.cls.__module__, self.cls.__name__, self.method)
 
-  def run(self):
-    return run_test(self.name(), self._run, config)
+    def run(self):
+        return run_test(self.name(), self._run, config)
 
-  def _run(self):
-    runner = Runner()
-    inst = self.cls(self.method)
-    test = getattr(inst, self.method)
+    def _run(self):
+        runner = Runner()
+        inst = self.cls(self.method)
+        test = getattr(inst, self.method)
 
-    if hasattr(inst, "configure"):
-      runner.run("configure", lambda: inst.configure(config))
-      if runner.halt(): return runner
-    if hasattr(inst, "setUp"):
-      runner.run("setup", inst.setUp)
-      if runner.halt(): return runner
+        if hasattr(inst, "configure"):
+            runner.run("configure", lambda: inst.configure(config))
+            if runner.halt():
+                return runner
+        if hasattr(inst, "setUp"):
+            runner.run("setup", inst.setUp)
+            if runner.halt():
+                return runner
 
-    runner.run("test", test)
+        runner.run("test", test)
 
-    if hasattr(inst, "tearDown"):
-      runner.run("teardown", inst.tearDown)
+        if hasattr(inst, "tearDown"):
+            runner.run("teardown", inst.tearDown)
 
-    return runner
+        return runner
+
+    def __repr__(self):
+        return "MethodTest(%r, %r)" % (self.cls, self.method)
 
-  def __repr__(self):
-    return "MethodTest(%r, %r)" % (self.cls, self.method)
 
 class PatternMatcher:
 
-  def __init__(self, *patterns):
-    self.patterns = patterns
+    def __init__(self, *patterns):
+        self.patterns = patterns
+
+    def matches(self, name):
+        for p in self.patterns:
+            if match(name, p):
+                return True
+        return False
 
-  def matches(self, name):
-    for p in self.patterns:
-      if match(name, p):
-        return True
-    return False
 
 class FunctionScanner(PatternMatcher):
 
-  def inspect(self, obj):
-    return type(obj) == types.FunctionType and self.matches(obj.__name__)
+    def inspect(self, obj):
+        return type(obj) == types.FunctionType and self.matches(obj.__name__)
+
+    def descend(self, func):
+        # the None is required for older versions of python
+        return
+        yield None
```

Comment:
@astitcher maybe something that can be also removed; marking this so it can be found in the future

Reply:
this thing is there multiple times

Reply:
Hmm, sure looks like it can go - best to check that the comment doesn't apply any more - what versions are 'older' for the purposes of this comment?

Reply:
No idea. Yield was added in Python 2.5 and yield without that `None` works for me in Python 2.7. So the "older versions" was either 2.5 or 2.6, or both, or maybe Jython?

Anyways, it is trying to create an empty generator function. After some googling, I think that this code can be replaced with

```python
def extract(self, obj):
    return ()
```

or, if we want to maintain the precise type,

```python
def extract(self, obj):
    return iter(())
```

This is better because return; yield can trigger warnings about unreachable code in some tooling.